development/webpack/utils/helpers.ts

```diff
@@ -0,0 +1,243 @@
+   * the browser extension platform itself.
+   *
+   * @param chunk
+   * @param chunk.name
+   * @returns
+   */
+  function canBeChunked({ name }: Chunk): boolean {
+    return !name || !selfContainedScripts.has(name);
+  }
+  return { entry, canBeChunked };
+}
+
+/**
+ * @param filename
+ * @param appRoot
+ * @throws Throws an `Error` if the file is an invalid entrypoint filename
+ * (a file starting with "_")
+ */
+function assertValidEntryFileName(filename: string, appRoot: string) {
+  if (!filename.startsWith('_')) {
+    return;
+  }
+
+  const relativeFile = relative(process.cwd(), join(appRoot, filename));
+  const error = `Invalid Entrypoint Filename Detected\nPath: ${relativeFile}`;
+  const reason = `Filenames at the root of the extension directory starting with "_" are reserved for use by the browser.`;
+  const newFile = filename.slice(1);
+  const solutions = [
+    `Rename this file to remove the underscore, e.g., '${filename}' to '${newFile}'`,
+    `Move this file to a subdirectory and, if necessary, add it manually to the build 😱`,
+  ];
+  const context = `This file was included in the build automatically by our build script, which adds all HTML files at the root of '${appRoot}'.`;
+
+  const message = `${error}
+  Reason: ${reason}
+
+  Suggested Actions:
+  ${solutions.map((solution) => ` •  ${solution}`).join('\n')}
+  ${`\n ${context}`}
+  `;
+
+  throw new Error(message);
+}
+
+/**
+ * It gets minimizers for the webpack build.
+ */
+export function getMinimizers() {
+  const TerserPlugin: typeof TerserPluginType = require('terser-webpack-plugin');
+  return [
+    new TerserPlugin({
+      // use SWC to minify (about 7x faster than Terser)
+      minify: TerserPlugin.swcMinify,
+      // do not minify snow.
+      exclude: /snow\.prod/u,
+    }),
+  ];
+}
+
+/**
+ * Helpers for logging to the console with color.
+ */
+export const { colors, toGreen, toOrange, toPurple } = ((depth, esc) => {
+  if (depth === 1) {
+    const echo = (message: string): string => message;
+    return { colors: false, toGreen: echo, toOrange: echo, toPurple: echo };
+  }
+  // 24: metamask green, 8: close to metamask green, 4: green
+  const green = { 24: '38;2;186;242;74', 8: '38;5;191', 4: '33' }[depth];
+  // 24: metamask orange, 8: close to metamask orange, 4: red :-(
+  const orange = { 24: '38;2;247;85;25', 8: '38;5;208', 4: '31' }[depth];
+  // 24: metamask purple, 8: close to metamask purple, 4: purple
+  const purple = { 24: '38;2;208;117;255', 8: '38;5;177', 4: '35' }[depth];
+  return {
+    colors: { green: `${esc}[1;${green}m`, orange: `${esc}[1;${orange}m` },
+    toGreen: (message: string) => `${esc}[1;${green}m${message}${esc}[0m`,
+    toOrange: (message: string) => `${esc}[1;${orange}m${message}${esc}[0m`,
+    toPurple: (message: string) => `${esc}[1;${purple}m${message}${esc}[0m`,
+  };
+})((process.stderr.getColorDepth?.() as 1 | 4 | 8 | 24) || 1, '\u001b');
```

Comment:
OK, so I code-golfed this a tiny bit.

Anyway, it lets us write colors on the terminal at different color bit depths (1, 4, 8, or 24)